tests/testthat/test-derive_var_shift.R

```diff
@@ -0,0 +1,125 @@
+    "P02",    "SODIUM",  147,   "Y",    "HIGH",  "HIGH"
+  )
+  expected_output <- tibble::tribble(
+    ~USUBJID, ~PARAMCD, ~AVAL, ~ABLFL, ~BNRIND, ~ANRIND, ~SHIFT1,
+    "P01",    "ALB",     33,    "Y",    "LOW",   "LOW",   "LOW to LOW",
+    "P01",    "ALB",     38,    "",     "LOW",   "NORMAL", "LOW to NORMAL",
+    "P02",    "ALB",     37,    "Y",    "NORMAL", "NORMAL", "NORMAL to NORMAL",
+    "P02",    "ALB",     49,    "",     "NORMAL", "HIGH",  "NORMAL to HIGH",
+    "P02",    "SODIUM",  147,   "Y",    "HIGH",  "HIGH",  "HIGH to HIGH"
+  )
+
+  expect_equal(
+    derive_var_shift(
+      input,
+      new_var = SHIFT1,
+      from_var = BNRIND,
+      to_var = ANRIND
+    ),
+    expected_output
+    )
+})
+
+
+test_that("Shift based on character variables with missing values", {
+  input <- tibble::tribble(
+    ~USUBJID, ~PARAMCD, ~AVAL, ~ABLFL, ~BNRIND, ~ANRIND,
+    "P01",    "ALB",     33,    "Y",    "LOW",   "LOW",
+    "P01",    "ALB",     38,    "",     "LOW",   "NORMAL",
+    "P01",    "ALB",     NA,    "",     "LOW",   "",
+    "P02",    "ALB",     NA,    "Y",    "",      "",
+    "P02",    "ALB",     49,    "",     "",      "HIGH",
+    "P02",    "SODIUM",  147,   "Y",    "HIGH",  "HIGH"
+  ) %>% convert_blanks_to_na
+  expected_output <- tibble::tribble(
+    ~USUBJID, ~PARAMCD, ~AVAL, ~ABLFL, ~BNRIND, ~ANRIND, ~SHIFT1,
+    "P01",    "ALB",     33,    "Y",    "LOW",   "LOW",   "LOW to LOW",
+    "P01",    "ALB",     38,    "",     "LOW",   "NORMAL", "LOW to NORMAL",
+    "P01",    "ALB",     NA,    "",     "LOW",   "",      "LOW to NULL",
+    "P02",    "ALB",     NA,    "Y",    "",      "",      "NULL to NULL",
+    "P02",    "ALB",     49,    "",     "",      "HIGH",  "NULL to HIGH",
+    "P02",    "SODIUM",  147,   "Y",    "HIGH",  "HIGH",  "HIGH to HIGH"
+  ) %>% convert_blanks_to_na
+
+  expect_equal(
+    derive_var_shift(
+      input,
+      new_var = SHIFT1,
+      from_var = BNRIND,
+      to_var = ANRIND
+    ),
+    expected_output
+  )
+})
+
+
+test_that("Shift based on numeric variables with missing values", {
+  input <- tibble::tribble(
+    ~USUBJID, ~PARAMCD, ~AVAL, ~ABLFL, ~BASE,
+    "P01",    "ALB",     33.1,  "Y",  33.1,
+    "P01",    "ALB",     38.5,  "",   33.1,
+    "P01",    "ALB",     NA,    "",   33.1,
+    "P02",    "ALB",     NA,    "Y",  NA,
+    "P02",    "ALB",     49.0,  "",   NA,
+    "P02",    "SODIUM",  147.5, "Y",  147.5
+  ) %>% convert_blanks_to_na
+  expected_output <- tibble::tribble(
+    ~USUBJID, ~PARAMCD, ~AVAL, ~ABLFL, ~BASE, ~SHIFT2,
+    "P01",    "ALB",     33.1,  "Y",  33.1,    "33.1 to 33.1",
+    "P01",    "ALB",     38.5,  "",   33.1,    "33.1 to 38.5",
+    "P01",    "ALB",     NA,    "",   33.1,    "33.1 to NULL",
+    "P02",    "ALB",     NA,    "Y",  NA,      "NULL to NULL",
+    "P02",    "ALB",     49,    "",   NA,      "NULL to 49",
+    "P02",    "SODIUM",  147.5, "Y",  147.5,   "147.5 to 147.5"
+  ) %>% convert_blanks_to_na
```

Comment:
Please directly create `NA` rather then `""` inside `tribble()`.

Reply:
Updated. I think I saw examples from other functions also creating tribbles using "" (e.g. `derive_var_pchg()`), are these minor or should we open an issue to clean up?